inst/include/interface/rcpp/rcpp_objects/rcpp_data.hpp

```diff
@@ -160,8 +160,8 @@ class AgeCompDataInterface : public DataInterfaceBase {
   template <typename Type>
   bool add_to_fims_tmb_internal() {
     std::shared_ptr<fims_data_object::DataObject<Type>> age_comp_data =
-        std::make_shared<fims_data_object::DataObject<Type>>(this->ymax,
-                                                             this->amax);
+        std::make_shared<fims_data_object::DataObject<Type>>(Type(this->ymax),
+                                                             Type(this->amax));
```

Comment:
There are five instances in this file where `this->*max` was wrapped in `Type(*)` erroneously. Or at least, it is causing the model to fail to compile and if I remove the changes in this file then FIMS compiles just fine. Thus, we either need documentation for why this change was made and how to get FIMS to compile with this change or the file needs to be reverted.